client/hotrod-client-transport/src/main/resources/hotrod.gr

```diff
@@ -0,0 +1,245 @@
+    | { UNKNOWN_VERSION_STATUS }? handleServerError { throw new HotRodClientException(serverMsg, messageId, status); }
+    | { COMMAND_TIMEOUT_STATUS }? serverMsg
+        {if (log.isTraceEnabled()) log.tracef("Server-side timeout performing operation: %s", serverMsg); }
+        { throw new HotRodClientException(serverMsg, messageId, status); }
+    | { ILLEGAL_LIFECYCLE_STATE }? serverMsg
+        { throw new RemoteIllegalLifecycleStateException(serverMsg, messageId, status); }
+    | { NODE_SUSPECTED }? serverMsg
+        { throw new RemoteNodeSuspectException(serverMsg, messageId, status); }
+    | { if (log.isTraceEnabled()) log.tracef("Unknown status: %#04x", status); }
+      { throw new IllegalStateException(String.format("Unknown status: %#04x", status)) }
+    ;
+
+handleServerError
+    : serverMsg
+      { Log.HOTROD.errorFromServer(serverMsg); }
+    ;
+
+// +----------+
+// |          |
+// | Topology |
+// |          |
+// +----------+
+newTopologyId: vInt;
+checkNewTopology
+    : { uByte == 1 }? newTopologyId topologyMembers hashFunctionVersion segmentOwners
+        { responseHandler.topologyUpdate(current(), newTopologyId, topologyMembers.toArray(new InetSocketAddress[0]), segmentOwners, hashFunctionVersion) }
+    | {}
+    ;
+
+hashFunctionVersion returns short
+    : { isHashDistributionAware(messageId) }? uByte
+    | { -1 }
+    ;
+numSegments: vInt;
+segmentOwners returns List<List<Integer>>
+    : { isHashDistributionAware(messageId) }?
+        numSegments { segmentOwners = allocList(numSegments) }
+            #numSegments ( ownersInSegments { segmentOwners.add(ownersInSegments); } )
+    | { null }
+    ;
+ownersInSegments returns List<Integer>
+    : uByte { ownersInSegments = allocList(uByte) }
+        #uByte (vInt { ownersInSegments.add(vInt); } )
+    ;
+
+clusterSize: vInt;
+topologyMembers returns List<InetSocketAddress>
+    : clusterSize { topologyMembers = allocList(clusterSize) }
+        #clusterSize (string uShort { topologyMembers.add(InetSocketAddress.createUnresolved(string, uShort)); } )
+    ;
+
+// +----------+
+// |          |
+// | Response |
+// |          |
+// +----------+
+handleResponseBody switch resOpCode
+    : { PING_RESPONSE }? keyType valueType negotiatedProtocolVersion serverOps
+                                 { responseHandler.pingResponse(current(), status, negotiatedProtocolVersion, keyType, valueType, serverOps); }
+    | { GET_RESPONSE }? getResponseBody { responseHandler.getResponse(current(), status, getResponseBody); }
+    | { PUT_RESPONSE }? assertOperationSuccess returnPossiblePrevValue { responseHandler.putResponse(current(), returnPossiblePrevValue, status); }
+    | { delegateParsing(buf, messageId, resOpCode, status); }
+    ;
+
+keyType: mediaType;
+valueType: mediaType;
+mediaType returns MediaType
+   : { delegate.currentProtocolVersionIsOlder(2, 9) >= 0 }? mediaTypeDefinition mediaTypeDescription { mediaTypeDescription }
```

Comment:
I believe the new parser only supports HR 4 right? Might as well rip out the old stuff at the same time.